pkg/controller/postgres/postgres_controller.go

```diff
@@ -0,0 +1,168 @@
+	err = c.Watch(&source.Kind{Type: &integreatlyv1alpha1.Postgres{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource Pods and requeue the owner Postgres
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &integreatlyv1alpha1.Postgres{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcilePostgres implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcilePostgres{}
+
+// ReconcilePostgres reconciles a Postgres object
+type ReconcilePostgres struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client       client.Client
+	scheme       *runtime.Scheme
+	logger       *logrus.Entry
+	ctx          context.Context
+	providerList []providers.PostgresProvider
+	cfgMgr       providers.ConfigManager
+}
+
+// Reconcile reads that state of the cluster for a Postgres object and makes changes based on the state read
+// and what is in the Postgres.Spec
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcilePostgres) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.logger.Info("Reconciling Postgres")
+
+	// Fetch the Postgres instance
+	instance := &integreatlyv1alpha1.Postgres{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	stratMap, err := r.cfgMgr.GetStrategyMappingForDeploymentType(r.ctx, instance.Spec.Type)
+	if err != nil {
+		return reconcile.Result{}, errorUtil.Wrapf(err, "failed to read deployment type config for deployment %s", instance.Spec.Type)
+	}
+	for _, p := range r.providerList {
+		if !p.SupportsStrategy(stratMap.Postgres) {
+			continue
+		}
+
+		// delete the postgres if the deletion timestamp exists
+		if instance.DeletionTimestamp != nil {
+			err := p.DeletePostgres(r.ctx, instance)
+			if err != nil {
+				return reconcile.Result{}, errorUtil.Wrapf(err, "failed to perform provider-specific storage deletion")
+			}
+			return reconcile.Result{}, nil
+		}
+
+		ps, err := p.CreatePostgres(r.ctx, instance)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if ps == nil {
+			r.logger.Info("Secret data is still reconciling, postgres instance is nil")
+			return reconcile.Result{}, errorUtil.New("secret data is still reconciling")
```

Comment:
Would it be worth removing this error here, and replacing it with something similar as we have done with Redis, 

`reconcile.Result{Requeue: true, RequeueAfter: time.Second * 30}, nil`

IMO the error makes it look like a problem while we are just waiting on the postgres instance to become available. We will catch any other errors in the provider. Open to suggestions, but we should make sure we follow the same pattern here across all the controllers.

Reply:
let's move to using the suggestion @ciaranRoche made here.

however, we still should have a way of communicating back to anyone who wants detail on the provision progress of the cloud resource. so i think it would make sense here to add a `phase` and `message` block to the status of our resources. `phase` being a one-word indicator of the cloud resources status and `message` being something like `"Waiting for Postgres resources to deploy"`.

we can do the status block updates in a separate pr. @ciaranRoche would you mind creating a ticket around that?

Reply:
Created that Jira -- https://issues.jboss.org/browse/INTLY-3705

Reply:
:+1: